src/js/sanitize.js

```diff
@@ -152,9 +152,9 @@ const FRACTIONS = Object.values(FRACT_MAP).join('');
 // Try to match things like "1 tablespoon sugar"
 const RECIPE_QUANTITY_RE = new RegExp([
   '^',
-  `((?:\\d+\\s?)?[\\d${FRACTIONS}⁄-]+)`,
+  `((?:\\.\\d+|\\d+(?:\\.\\d+)?)?(?:\\s*[${FRACTIONS}-])?(?:\\s*⁄\\d+)?)`,
```

Comment:
Can split decimals and fractions to separate regexes here to make it slightly more readable. Although with how many edge cases have been added over the years I wouldn't be surprised if someone on the internet used `.5 ¼ cloves`

Reply:
I always start with the least-invasive / most minimal approach that passes all the existing tests. Having said that, regexes are a write-only medium. The subsequent code assumes 3 capture groups, and I was trying to preserve that. All this code lives within the *quantity* capture group.

Within this, there's three non-capturing groups in sequence, but each non-capturing group is essentially optional, and independent. Any subset of them can match, and you'll get a match object back.

The first group handles numbers of the following forms: `.y` `x`, `x.y` where `x` and `y` are arbitrary sequences of at least one digit.

The second group handles fractions. The the third group was intended to deal with trailing numbers. I think this might be a comment on an older version of the code, because I thought I handled the quote character in that sub-pattern.

Reply:
I see what you're saying. This is potentially too permissive. You may be right.

And what you're really suggesting is to change the nesting of the capture groups so, in pseudo-bnf:

   quantity => Integer? ( '.' Integer | '\'' Integer | Fraction)? 

Vs what I have

   quantity => (('.' Integer) | Integer ('.' Integer))?(Fraction)?(Quote)?`

You might be right. I guess it depends what you want. I am biased toward permissiveness here, but I could see it the other way around also.

Reply:
Honestly permissiveness is generally good here, internet recipes are an endless stream of edge cases. The original regex also allows some invalid sequences anyway. My preference was mainly that the change hit some inflection point in complexity where I don't think I would make sense of the regex next time I see it.

Reply:
Well yeah, regexes are notorious, and what's worse you can't include inline comments.

I think I'll leave this one as-is, and could see doing a more in-depth refactoring of this code into something more akin to a recursive-descent parser. This would allow breaking this up into smaller, more comprehensible units. And could support alternative forms like `Ingredient: Quantity Unit`, etc.

But I think I would need to add a lot more test cases. Frankly I'm impressed it works as well as it does.

Reply:
I thought about it some more, and I think within one regex it has to be this way, in order to handle the leading decimal point. The first example is equivalent to the original regex.